garden-backend-service/src/api/tasks/auto_deletion/utils.py

```diff
@@ -49,13 +91,51 @@ async def mark_gardens_for_deletion(session_maker):
 
 
 async def mark_modal_apps_for_deletion(session_maker) -> int:
-    # First get all modal function IDs that are used in gardens
-    used_function_ids = select(gardens_modal_functions.c.modal_function_id)
+    # First get all modal function IDs that are used in published gardens
+    used_function_ids = (
+        select(gardens_modal_functions.c.modal_function_id)
+        .join(Garden, Garden.id == gardens_modal_functions.c.garden_id)
+        .where(Garden.doi_is_draft.isnot(True))
```

Comment:
I think this implies that we won't delete apps used in archived gardens. I think we also want to delete apps where doi_is_draft is False, but is_archived is True. (ie, the garden is archived)